design-documents/graph-ql/coverage/company-credit.md

```diff
@@ -0,0 +1,39 @@
+# Queries
+
+```graphql
+
+###### Begin: Extending existing types ######
+type Company {
+    credit: CompanyCredit! @doc(description: "Company credit balance")
+    credit_history: CompanyCreditHistory! @doc(description: "Company credit operations history")
+}
+###### End: Extending existing types ######
+
+
+###### Begin: Defining new types ######
+type CompanyCreditHistory {
+    items: [CompanyCreditOperation]! @doc(description: "An array of company credit operations")
+    page_info: SearchResultPageInfo! @doc(description: "Metadata for pagination rendering")
+}
+
+type CompanyCreditOperation {
+    date: String! @doc(description: "The date of the company credit operation")
+    # Operation may be of enum type
+    operation: String! @doc(description: "The type of the company credit operation")
+    amount: Money @doc(description: "The amount fo the company credit operation")
+    balance: CompanyCredit! @doc(description: "Credit balance after the company credit operation")
+    purchase_order: String @doc(description: "Purchase order number associated with the company credit operation")
+    updated_by: String! @doc(description: "The name of the person submitting the company credit operation")
```

Comment:
The change in credit balance could be triggered from admin or storefront part. Thus it might be unclear which side or which user did a change. I think we can add user type to this filed or split this field on two(or separate type): username & user type

Reply:
Do we have User type stored in history? It is not currently visible on the storefront in customer account.

Reply:
Yes, we have it stored. View level suppress this information :
```
$userName = '';
        if ($userType == UserContextInterface::USER_TYPE_ADMIN) {
            $userName = $this->userFactory->create()->load($key)->getName();
        } elseif ($userType == UserContextInterface::USER_TYPE_CUSTOMER) {
            $user = $this->customerRepository->getById($key);
            $userName = $this->nameGeneration->getCustomerName($user);
        }
```
For me it make sense to have it in API, but feel free to skip this comment. We can add user type later

Reply:
Added after discussion with Nishant.